lnd.go

```diff
@@ -343,6 +332,66 @@ func Main(cfg *Config, lisCfg ListenerCfg, interceptor signal.Interceptor) error
 		}
 	}
 
+	// Start leader election if we're running on etcd. Continuation will be
+	// blocked until this instance is elected as the current leader or
+	// shutting down.
+	elected := false
+	if cfg.Cluster.EnableLeaderElection &&
+		cfg.Cluster.LeaderElector == cluster.EtcdLeaderElector {
```

Comment:
If we're going with the factory approach, do we need this additional check here? The config validation should've take care of this assertion already, so we don't really care what kind of election we're running here.

Reply:
Instead we could perhaps log what kind of election is used?

Reply:
done